docs/readmes/proposals/p024_magma_settlement_service.md

```diff
@@ -0,0 +1,280 @@
+
+### 3.3 Integration interfaces
+
+The following figure shows the integration interfaces between Magma domain and OCS /DCH.
+
+![Integration interfaces](https://user-images.githubusercontent.com/93994458/149514277-062e0834-cb8c-49af-b0e4-867c4664c694.png)
+
+#### 3.3.1 DIAMETER Gy interface specification
+
+Interface transport protocol: TCP/Diameter/DCCA.
+The 3GPP based Gy interface between Session Proxy and Session Purchaser to implement real-time quota management for the packet data service for the roaming subscribers. While this interface resembles Gy in almost every way, this is a non-standard use of the Gy protocol defined to allow accounting at the operator level and allow the session purchaser a control point to approve or deny sessions at initiation or at update points throughout a session.
+The Session Purchaser is the Diameter Credit Control server, which provides the online charging data to the Session Proxy based on volume quota.
+The connection between the Session Proxy (client) and Session Purchaser (server) is TCP based. There are a series of message exchanges to manage quota in real-time.
+
+#### 3.3.2 MME/SGW ↔︎ Helium Miner
+
+Session lifecycle events and usage reports should be delivered from each deployed AGW to the Network-Server (helium-handler), which uses that information to update a state channel. The Helium Miner container does such communication between hotspot(AGW) and Network Server. So, a trusted channel should be formed between MME/SGW and Miner to let a Miner container communicate with the Helium network on behalf of AGW. For having a trusted channel between AGW and Miner new component is introduced - a session-forwarder. Session-forwarder is a mediator between Magma-specific components deployed on AGW and a Helium Miner.
+To let Miner trust Session-forwarder, session-forwarder is deployed on the same AGW where software is signed and Linux Secure Boot is used to verify all software components being loaded. So, Miner can trust to all received session-related messages. Miner just does forward all message to helium-handler.
+
+![SGW-Miner](https://user-images.githubusercontent.com/93994458/149515220-2ed6ea77-bcb3-4322-a615-c23abc54f580.png)
+
+1. Sessiond(Magma component) requires new sessions statistics gRPC stream API.
+2. When new session is created or SessionState is updated on sessiond side, sessiond streams those updates via session statistics API to session-forwarder.
+3. Session-forwarder talks to helium miner via gRPC.
+
+Sessiond API update:
+`service SessionStaticsStreamer {
+rpc SessionStatistics(orc8r.Void) returns (stream SessionStatisticsUpdate) {}
+}`
+
+Miner API update (session-forwarder <-> Miner) :
+`service api {
+rpc session_usage(session_usage_update) returns (orc8r.Void) {}
+}`
+
+#### 3.3.3 Helium Miner ↔︎ Network Server (router/helium-handler)
+
+For each 3gpp session Miner should forward session usage stats from Session-forwarder to Network Server.
+`service api {
```

Comment:
Multi-line block quotes (here and elsewhere in this document) are handled by [Fenced Code Blocks](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks). This will improve readability of what is otherwise displayed as a single line.

Reply:
agree. done